### PR TITLE
ADDED: SANs, show serial#/fingerprints for leaf/intermediaries/root c…

### PR DESCRIPTION
**Pull request by elvischang-lll**
ADDED: more features - SANs, display all the certificates(leaf/intermediaries/root) serial#/fingerprints...and general cleanups....also added new color and color functions for future.

### CODE REVIEW

File: ssl [hunk 1]
```diff
@@ -72,59 +91,54 @@ $(ColorGreen './ssl domain.com')
...
-        echo -ne "$(ColorRed '#') $(ColorGreen 'Certificate Expires In:')"
-	echo ''
-
+   echo -e "$(ColorRed '#') $(ColorGreen 'Leaf Certificate Expires In:')"
 	# macos - format
 	#date -j -f "%b %d %T %Y %Z" "$expires" +'%s'
 	#date -j -f "%F" "$today" +'%s'
 	## echo $(( ( $(date -ud "$expires" +'%s') - $(date -ud "$today" +'%s') )/60/60/24 )) days
-	echo $(( ( $(date -j -f "%b %d %T %Y %Z" "$expires" +'%s') - $(date -j -f "%F" "$today" +'%s') )/60/60/24 )) days
+	echo "	$(( ( $(date -j -f "%b %d %T %Y %Z" "$expires" +'%s') - $(date -j -f "%F" "$today" +'%s') )/60/60/24 )) days"
```
Comment (bobbyiliev) [inline ssl]: That looks great! Thank you for your contribution.

It looks like that on line 105 the `-j`  argument is throwing the following error:

```
date: invalid option -- 'j'
Try 'date --help' for more information.
date: invalid option -- 'j'
Try 'date --help' for more information.
ssl3: line 105: (  -  )/60/60/24 : syntax error: operand expected (error token is ")/60/60/24 ")
```

Comment (elvischang-lll) [reply ssl]: Forgot to put in checks for OS version...ie macos/linux....am putting in checks for that soon.

**Review by elvischang-lll — COMMENTED** [review]
Added: OS version checks for DATE command for different OSs.

**Review by bobbyiliev — APPROVED** [review]
Looks awesome! Thank you for the contribution! 🙌

**Pull request merged**